app/src/main/cpp/dictionary.cpp

```diff
@@ -72,7 +72,10 @@ int Dictionary::getSuggestions(int *codes, int codesSize, unsigned short *outWor
 
     // Get the word count
     suggWords = 0;
-    while (suggWords < mMaxWords && mFrequencies[suggWords] > 0) suggWords++;
+    while (suggWords < mMaxWords && sizeof(mFrequencies) < suggWords-1 &&
```

Comment:
This doesn't work. mFrequencies is an int*, so sizeof(mFrequencies) is just the size of a pointer (4 on a 32bit platform).

If it were an int array with a compile-time known size, you'd need to use sizeof(mFrequencies)/sizeof(mFrequencies[0]) or similar to get the element count as opposed to the number of bytes used.

Reply:
Hi @klausw , this PR was part of a security issue I am trying to discuss with hackerskeyboard developers at  "pocketworkstation@gmail.com". As I did not get any response from that email, I created this PR to get it fixed. If you could share a communication medium where I can use and explain about this security issue, we can discuss on how to properly fix it.